packages/lesswrong/components/users/DialogueMatchingPage.tsx

```diff
@@ -0,0 +1,805 @@
+          userId
+          targetUserId
+          checked
+          match
+        }
+      }
+    `)
+
+  async function handleNewMatchAnonymisedAnalytics() {
+    captureEvent("newDialogueReciprocityMatch", {}) // we only capture match metadata and don't pass anything else
+
+    // ping the slack webhook to inform team of match. YOLO:ing and putting this on the client. Seems fine: but it's the second time this happens, and if we're doing it a third time, I'll properly move it all to the server 
+    const webhookURL = "https://hooks.slack.com/triggers/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";
+    const data = {} // Not sending any data for now 
+    void pingSlackWebhook(webhookURL, data)
+  }
+
+  const [showConfetti, setShowConfetti] = useState(false);
+
+
+  async function updateDatabase(event: React.ChangeEvent<HTMLInputElement>, targetUserId: string, checkId?: string) {
+    if (!currentUser) return;
+
+    const response = await upsertDialogueCheck({
+      variables: {
+        targetUserId: targetUserId, 
+        checked: event.target.checked
+      },
+      update(cache, { data }) {
+        if (!checkId) {
+          cache.modify({
+            fields: {
+              getUsersDialogueChecks(existingChecksRefs = []) {
+                const newCheckRef = cache.writeFragment({
+                  data: data.upsertUserDialogueCheck,
+                  fragment: gql`
+                    fragment NewCheck on DialogueChecks {
+                      _id
+                      __typename
+                      userId
+                      targetUserId
+                      checked
+                      match
+                    }
+                  `
+                });
+                const newData = [...existingChecksRefs, newCheckRef]
+                return newData;
+              }
+            }
+          });
+        }
+      },
+      optimisticResponse: {
+        upsertUserDialogueCheck: {
+          _id: checkId || randomId(),
+          __typename: 'DialogueCheck',
+          userId: currentUser._id,
+          targetUserId: targetUserId,
+          checked: event.target.checked,
+          match: false 
+        }
+      }
+    })
+    
+    if (response.data.upsertUserDialogueCheck.match) {
+      void handleNewMatchAnonymisedAnalytics()
+      setShowConfetti(true);
+      setTimeout(() => setShowConfetti(false), 4000); // Reset the variable after 5 so it can be clicked again
+    }
+  }
+
+  return (
+    <>
+      {showConfetti && <ReactConfetti recycle={false} colors={["#7faf83", "#00000038" ]} />}
+      <FormControlLabel
+        control={ 
+          <Checkbox 
+            classes={{
+              root: classNames({
+                [classes.checkbox]: !isChecked,
+                [classes.checkboxCheckedMatched]: isChecked && isMatched,
+                [classes.checkboxCheckedNotMatched]: isChecked && !isMatched
+              }),
+              checked: classes.checked
```

Comment:
Just noticed this whole thing is in a `classes` block rather than being assigned to the `className` attribute.  Idk exactly why/what it's doing differently, but add it to the list of things to fix in a follow-up